.ci/docker/common/install_executorch.sh

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -ex
+
+source "$(dirname "${BASH_SOURCE[0]}")/common_utils.sh"
+
+clone_executorch() {
+  EXECUTORCH_PINNED_COMMIT=$(get_pinned_commit executorch)
+
+  # Clone the Executorch
+  git clone https://github.com/pytorch/executorch.git
+
+  # and fetch the target commit
+  pushd executorch
+  git checkout "${EXECUTORCH_PINNED_COMMIT}"
+  git submodule update --init
+  popd
+
+  chown -R jenkins executorch
+}
+
+install_buck2() {
+  pushd executorch/.ci/docker
+
+  BUCK2_VERSION=$(cat ci_commit_pins/buck2.txt)
+  source common/install_buck.sh
+
+  popd
+}
+
+install_conda_dependencies() {
+  pushd executorch/.ci/docker
+  # Install conda dependencies like flatbuffer
+  conda_install --file conda-env-ci.txt
+  popd
+}
+
+install_pip_dependencies() {
+  pushd executorch/.ci/docker
+
+  # Install all Python dependencies
+  pip_install -r requirements-ci.txt
+  # NB: Just need to install the stable version of these domain libraries here
+  # so that we can install executorch.  They will need to be recompiled later
+  # together with torch
+  pip_install torchvision torchaudio
+
+  popd
+}
+
+setup_executorch() {
+  pushd executorch
+  source .ci/scripts/utils.sh
```

Comment:
This job will rely on some config/scripts in the executorch repo to set up itself. Installing packages and dependencies may be fine but the scripts can be tricky. My concern is that if there are changes in those scripts in ET and it won't be aware of that there is a job in PT repo relies on it, how can we ensure the ET job in PT won't break if someone mess up the scripts/config? It seems like a circular dependency issue

Reply:
True, the reason I use this script is because I don't think it's a good idea to copy/paste the logic from ET CI over here. We have a safe guard though, if someone unknowingly break the script and commit it in ET, next time the pin commit is update, the job here would fail blocking it from landing in PT trunk.  This is the same pattern we are using with XLA here https://github.com/pytorch/pytorch/blob/main/.ci/pytorch/test.sh#L823 where PT CI loads the `common.sh` script from XLA repo.  I guess the solution here is to make sure that `.ci/scripts/utils.sh` stays where it is atm, and backward compatible

Reply:
So specifically, I need these 2 function from the script `install_flatc_from_source` and `build_executorch_runner "buck2"`.  Do you think it's better to copy them here instead of trying to source the `utils.sh`?

Reply:
I agree with you that current setting is better than copy/paste the logic from ET CI, that would out of maintenance over time. `install_flatc_from_source ` and `build_executorch_runner "buck2"` are relatively stable and testable in ET, I'm less concerning about it. I guess my key questions here are:
  1. Is there any chance the ET job fails in PT due to setup issues (e.g. missing packages, incompatible versions, etc.)?
  2. What is the mitigation process/steps when #1 happens to maximize the availability of the ET job? For example, it would be pretty bad if the fix must be landed to ET first and PT has to wait for a day so that the fix will be picked by the ET nightly build. BC breaking changes could be landed during the downtime.

Reply:
Ah no, PT will checkout and build ET pinned commit directly, there will be no nightly delay.  ET pinned commit here is updated in a controlled way via PR either manual or automated update.  On the automated route, we could have alert if it starts failing, so oncall could address the issue as soon as possible.  On the manual route, whoever update the pinned commit will be blocked if the change on ET break the tests here.